src/lib.rs

```diff
@@ -129,10 +128,77 @@ fn parse_dependency(s: &str) -> Dependency {
         let mut it = s.split('+').map(|x| x.to_string());
         let name = it.next().unwrap();
         let features = it.collect::<Vec<_>>();
+
+        if name.ends_with(".git") {
+            let url = name.clone();
+            let name = std::path::Path::new(&name)
+                .file_stem()
+                .expect("cannot parse repo name")
+                .to_str()
+                .expect("cannot convert repo name")
+                .to_string();
+
+            Dependency::Repository {
+                name,
+                url,
+                branch: None,
+                rev: None,
+                features,
+            }
+        } else {
+            Dependency::CrateIo {
+                name,
+                version: None,
+                features,
+            }
+        }
+    }
+}
+
+pub fn format_dependency(dependency: &Dependency) -> String {
```

Comment:
If you don't call it it's basically dead code.

(Calling it in tests doesn't count 😁 )

Reply:
https://github.com/yozhgoor/cargo-temp/pull/46/files#diff-1d06da761111802c793c6e5ca704bfa0d6336d0becf87fddff02d81548a838abR98